File: backend/internal/models/policy.go
package models

import "time"

type Policy struct {
	ID        ID             `bson:"_id,omitempty" json:"-"`
	JSONID    JSONID         `bson:"-" json:"id"`
	Name      string         `bson:"name" json:"name"`
	Version   int            `bson:"version" json:"version"`
	Policy    map[string]any `bson:"policy" json:"policy"`
	CreatedAt time.Time      `bson:"createdAt" json:"createdAt"`
	UpdatedAt time.Time      `bson:"updatedAt" json:"updatedAt"`
}

func (p Policy) WithJSONID() Policy {
	p.JSONID = NewJSONID(p.ID)
	return p
}

File: frontend/src/types/policy.ts
export type Policy = {
  id: string
  name: string
  version: number
  policy: Record<string, unknown>
  createdAt: string
  updatedAt: string
}

export type CreatePolicyRequest = {
  name: string
  policy: Record<string, unknown>
}

export type UpdatePolicyRequest = {
  name?: string
  policy?: Record<string, unknown>
}
